File: Core/src/main/java/jokatu/game/stage/AnyEventSingleInputStage.java
package jokatu.game.stage;

import jokatu.game.event.GameEvent;
import jokatu.game.input.Input;
import jokatu.game.player.Player;

/**
 * A {@link SingleInputStage} that can fire any event.
 */
public abstract class AnyEventSingleInputStage<I extends Input, P extends Player> extends SingleInputStage<I, P, GameEvent> {
}

File: Core/src/main/kotlin/jokatu/game/stage/AnyEventSingleInputStage.kt
package jokatu.game.stage

import jokatu.game.event.GameEvent
import jokatu.game.input.Input
import jokatu.game.player.Player
import kotlin.reflect.KClass

/**
 * A [SingleInputStage] that can fire any event.
 */
abstract class AnyEventSingleInputStage<I : Input, P : Player>(
		inputClass: KClass<I>,
		playerClass: KClass<P>
) : SingleInputStage<I, P, GameEvent>(inputClass, playerClass)
